File: JavaAndScala/src/main/java/com/java/se/conclusion/design/factory/abstracts/BrandFactory.java
package com.java.se.conclusion.design.factory.abstracts;

import com.java.se.conclusion.design.factory.abstracts.common.Border;
import com.java.se.conclusion.design.factory.abstracts.common.Button;

/**
 * 	This is an interface to implement abstract factory pattern
 *  -- This interface must be implemented by its child classes
 *  
 * @author VinceYuan
 *
 */
public interface BrandFactory {

	public Button getButton();
	public Border getBorder();
}

File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/design/factory/abstracts/BrandFactory.scala
package org.lang.scala.conclusion.design.factory.abstracts

import org.lang.scala.conclusion.design.factory.abstracts.common.Border
import org.lang.scala.conclusion.design.factory.abstracts.common.Button

/**
 * 	This is a trait to implement abstract factory pattern
 *  -- This trait must be implemented by its child classes
 *  
 * 	@author VinceYuan
 */
trait BrandFactory {
  
  def getButton(): Button
  def getBorder(): Border
}